test/node/error.test.ts

```diff
@@ -102,4 +108,24 @@ describe('BSONError', function () {
       expect(new BSONRuntimeError('Woops!')).to.have.property('name', 'BSONRuntimeError');
     });
   });
+
+  describe('class BSONOffsetError', () => {
+    it('is a BSONError instance', function () {
+      expect(BSONError.isBSONError(new onDemand.BSONOffsetError('Oopsie', 3))).to.be.true;
+    });
+
+    it('has a name property equal to "BSONOffsetError"', function () {
+      expect(new onDemand.BSONOffsetError('Woops!', 3)).to.have.property('name', 'BSONOffsetError');
+    });
+
+    it('sets the offset property', function () {
+      expect(new onDemand.BSONOffsetError('Woops!', 3)).to.have.property('offset', 3);
+    });
+
+    it('includes the offset in the message', function () {
+      expect(new onDemand.BSONOffsetError('Woops!', 3))
+        .to.have.property('message')
+        .that.matches(/offset: 3/i);
```

Comment:
What's the function of the `/i `at the end of a lot of these expect statements / where is it coming from ?

Reply:
That's a regular expression flag. It makes the search case insensitive. There are other regular expression flags [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_expressions#advanced_searching_with_flags)

Reply:
Good to know!